app/controllers/projects_controller.rb

```diff
@@ -13,19 +13,19 @@ def create
   end
 
   def new
-    @project = Project.new
+    @project = authorize Project.new
   end
 
   def show
-    @project = Project.find_by(id: params[:id])
+    @project = authorize Project.find_by(id: params[:id])
   end
 
   def edit
-    @project = Project.find_by(id: params[:id])
+    @project = authorize Project.find_by(id: params[:id])
   end
 
   def update
-    @project = Project.find(params[:id])
+    @project = authorize Project.find(params[:id])
```

Comment:
Érdemes lehet egy `before_action` -be kiszervezni ezeket az authorize hívásokat.
Consider the following:

```
class AsdController < AppController
  before_action :load_asd, only: %i[show update]

  def show
  end
  
  def update
    if @asd.update(asds_params)
    # imagine here the rest of the update logic
  end
  
  private
  
  def load_asd
    @asd = authorize Asd.find_by(id: params[:id])
  end
end
```

Van egy `Asd` kontrollerem. Ennek a kontrollernek bizonyos akcióihoz ugyanazt az egy sor kódot másolnám le: authorizáljuk a params id-jában érkező id-jú Asd-ot. E helyett azt tudom mondani, hogy ezt a kódot kiszervezem egy külön privát függvénybe. És bizonyos akciók előtt meghívom. Legyen a függvény neve `load_asd`. Ilyenkor, ha beérkezik egy kérés a `show` endpoint-ra, először le fog futni a `load_asd` (ha az `only: []` részben megadtam ezt az akciót), ezért egy kukacos `@asd` változóban elérhető lesz számomra a keresett Asd, és a `show` függvénybe nem kell semmit írnom. Nincs kódduplikáció.

Érdemes megjegyezni, hogy ezzel a segédfüggvénnyel mindig az adott policy adott endpoint-hoz kapcsolódó helpere fog meghívódni. Szóval ha a `show` endpoint előtt hívom meg a `load_asd` -ot, akkor a `show?` policy helper fug futni, ha az `update` előtt hívom meg a `load_asd` -ot, akkor az `update?` fog meghívódni. Pont mint ahogy kényelmes.